apps/cowswap-frontend/src/modules/hooksStore/dapps/WithdrawCoWAMMApp/pure/PoolBalancesPreview/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { TokenLogo } from '@cowprotocol/tokens'
+import { TokenAmount } from '@cowprotocol/ui'
+import { CurrencyAmount, Fraction } from '@uniswap/sdk-core'
+
+import { Label } from 'modules/hooksStore/styled'
+
+import { FiatValue } from 'common/pure/FiatValue'
+
+import * as styledEl from './styled'
+
+import { PoolBalance } from '../../types'
+
+export interface PoolBalancesProps {
+  poolBalance: PoolBalance[]
+  label: string
+  backgroundColor?: string
+}
+
+export interface PoolBalancePreviewProps {
+  id: string
+  poolBalance: PoolBalance
+}
+
+export function PoolBalancesPreview(props: PoolBalancesProps) {
+  const { poolBalance, label, backgroundColor } = props
+
+  return (
+    <div>
+      <div>
+        <Label>{label}</Label>
+      </div>
+      <styledEl.Wrapper backgroundColor={backgroundColor}>
+        {poolBalance.map((poolBalance, index) => (
+          <PoolBalancePreview key={index} id={`${label}-${index}`} poolBalance={poolBalance} />
+        ))}
```

Comment:
I think you should avoid using the index as key, in many cases it can cause a lot of trouble.
https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/